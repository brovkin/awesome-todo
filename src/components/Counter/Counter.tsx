import React, { FC } from 'react';
import { decrement, increment } from '../../features/counter/counterSlice';
import { useAppDispatch, useAppSelector } from '../../app/hooks';

const Counter: FC = () => {
  const counter = useAppSelector((state) => state.counter.value);
  const dispatch = useAppDispatch();

  return (
    <>
      <div>Counter: {counter}</div>
      <button onClick={() => dispatch(increment())}>Inc</button>
      <button onClick={() => dispatch(decrement())}>Dec</button>
    </>
  );
};

export default Counter;
