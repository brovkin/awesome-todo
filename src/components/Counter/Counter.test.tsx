import React from 'react';
import configureStore from 'redux-mock-store';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import '@testing-library/jest-dom';
import Counter from './Counter';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('Counter Component', () => {
  let store;
  let component: ReactTestRenderer;

  beforeEach(() => {
    store = mockStore({
      counter: {
        value: 0,
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <Counter />
      </Provider>
    );
  });
  test('snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
