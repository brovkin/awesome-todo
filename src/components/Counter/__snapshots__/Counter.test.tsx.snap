// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component snapshot 1`] = `
Array [
  <div>
    Counter: 
    0
  </div>,
  <button
    className="inc"
    onClick={[Function]}
  >
    Inc
  </button>,
  <button
    className="dec"
    onClick={[Function]}
  >
    Dec
  </button>,
]
`;
